client/aws/aws.go

```diff
@@ -28,27 +28,29 @@ type rawJSON struct {
 	Term    interface{} `json:"terms"`
 }
 
-type EngineType string
+// engineType is the engine type specified in AWS api message.
+// we implement its String() method to convert it into the type stored in ours.
+type engineType string
```

Comment:
Should be capitalized.

normally only built-in type like string is not capitalized (and it's a way to figure out which one is built-in type)